R/individuals.R

```diff
@@ -0,0 +1,185 @@
+                  states$IOxGetLive1,
+                  states$IOxGetLive2,
+                  states$IOxGetDie1,
+                  states$IOxGetDie2,
+                  states$IOxNotGetLive1,
+                  states$IOxNotGetLive2,
+                  states$IOxNotGetDie1,
+                  states$IOxNotGetDie2,
+                  states$IMVGetLive1,
+                  states$IMVGetLive2,
+                  states$IMVGetDie1,
+                  states$IMVGetDie2,
+                  states$IMVNotGetLive1,
+                  states$IMVNotGetLive2,
+                  states$IMVNotGetDie1,
+                  states$IMVNotGetDie2,
+                  states$IRec1,
+                  states$IRec2,
+                  states$R,
+                  states$D),
+    variables = list(),
+    events = list()
+  )
+
+  list(human = human)
+}
+
+#' @title probabilities of the states
+#'
+#' @param dt time step
+#' @param psq parameters
+#'
+#' @return states
+#' @export
+Probabilities_of_states <- function(dt, psq) {
+
+  # probabilities
+  pstates <- list(pgamma_E = 1 - exp(-1.0 * (psq$gamma_E * dt)),
+    pgamma_IMild = 1 - exp(-psq$gamma_IMild * dt),
+    pgamma_ICase = 1 - exp(-1.0 * (psq$gamma_ICase * dt)),
+    pgamma_get_ox_survive = 1 - exp(-1.0 * (psq$gamma_get_ox_survive * dt)),
+    pgamma_not_get_ox_survive =
+      1 - exp(-1.0 * (psq$gamma_not_get_ox_survive * dt)),
+    pgamma_get_ox_die = 1 - exp(-1.0 * (psq$gamma_get_ox_die * dt)),
+    pgamma_not_get_ox_die = 1 - exp(-1.0 * (psq$gamma_not_get_ox_die * dt)),
+    pgamma_get_mv_survive = 1 - exp(-1.0 * (psq$gamma_get_mv_survive * dt)),
+    pgamma_not_get_mv_survive =
+      1 - exp(-1.0 * (psq$gamma_not_get_mv_survive * dt)),
+    pgamma_get_mv_die = 1 - exp(-1.0 * (psq$gamma_get_mv_die * dt)),
+    pgamma_not_get_mv_die  = 1 - exp(-1.0 * (psq$gamma_not_get_mv_die * dt)),
+    pgamma_rec  = 1 - exp(-1.0 * (psq$gamma_rec * dt)))
+
+  pstates
+}
+
+#' @title Continuous age variable
+#' @description Create a continuous age variable for the population
+#'
+#' @param pop population list
+#'
+#' @return continuous age variable
+#' @export
+#'
+#' @importFrom stats dexp
+Create_continuous_age_variable <- function(pop) {
+
+  r <- list()
+  for(i in 1:(85/5)){
+    r[[i]] <- (1 + ((i-1) * 5)) : (5*i)
+  }
+
+  ages <- list()
+  for(i in seq_len(length(pop$age_group))) {
+    ages[[i]] <- sample(r[[i]], pop$n[i], replace = TRUE,
+                        prob = dexp(r[[i]], 1/(21)))
+  }
+
+  ages <- unlist(ages)
+  return(ages)
+}
+
+#' @title Discrete age variable
+#' @description Create a discrete age variable for each of the
+#' length(pop$age_group) distinct age groups
+#'
+#' @param pop population list
+#'
+#' @return discrete age variable
+#' @export
+Create_discrete_age_variable <- function(pop) {
+
+  age_cont <- Create_continuous_age_variable(pop)
+
+  age_bins <- levels(cut(age_cont, breaks = c(seq(0, 80, 5), 999),
+                         include.lowest = TRUE))
+
+  # TO DO
```

Comment:
Can we throw an error? I could accidently use this and not know why it's not working

```suggestion
stop("Not yet implemented")
```